components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1084,20 +868,351 @@ subroutine dropmixnuc( &
+               !          qact_ss = fcloud*fact*qtot
+               !       where fcloud is cloud fraction, fact is activation fraction,
+               !       qtot=qact+qint, qint is interstitial particle mixratio
+               !    the activation rate (from mixing within the layer) can now be
+               !       written as
+               !          d(qact)/dt = (qact_ss - qact)/taumix_internal
+               !                     = qtot*(fcloud*fact*wtke/dz) - qact*(wtke/dz)
+               !    note that (fcloud*fact*wtke/dz) is equal to the nact/mact
+               !    also, d(qact)/dt can be negative.  in the code below
+               !       it is forced to be >= 0
+               !
+               ! steve --
+               !    you will likely want to change this.  i did not really understand
+               !       what was previously being done in k=pver
+               !    in the cam3_5_3 code, wtke(i,pver) appears to be equal to the
+               !       droplet deposition velocity which is quite small
+               !    in the cam3_5_37 version, wtke is done differently and is much
+               !       larger in k=pver, so the activation is stronger there
+               !
+
+               do imode = 1, ntot_amode
+                  mm = mam_idx(imode,0)
+                  fluxn(imode) = fluxn(imode)*delz_cld
+                  fluxm(imode) = fluxm(imode)*delz_cld
+                  nact(kk,imode) = nact(kk,imode) + fluxn(imode)*crdz
+                  mact(kk,imode) = mact(kk,imode) + fluxm(imode)*crdz
+                     ! note that kp1 is used here
+                  fluxntot = fluxntot &
+                        + fluxn(imode)*raercol_nsav(kp1,mm)*cs_col_in(kk)
+               enddo
+               nsource_col(kk) = nsource_col(kk) + fluxntot/(cs_col_in(kk)*dz(kk))
+
+            endif  ! (cldn(icol,kk) - cldn(icol,kp1) > 0.01)
+
+         else
+!  if cldn < 0.01_r8 at any level except kk=pver, deplete qcld, turn all raercol_cw to raercol, put appropriate tendency
+!  in nsource
+!  Note that if cldn(kk) >= 0.01_r8 but cldn(kk) - cldn(kp1)  <= 0.01, nothing is done.
+
+            ! no cloud
+
+            nsource_col(kk) = nsource_col(kk) - qcld(kk)*dtinv
+            qcld(kk)      = 0
+
+            ! convert activated aerosol to interstitial in decaying cloud
+
+            do imode = 1, ntot_amode
+               mm = mam_idx(imode,0)
+               raercol_nsav(kk,mm)    = raercol_nsav(kk,mm) + raercol_cw_nsav(kk,mm)  ! cloud-borne aerosol
+               raercol_cw_nsav(kk,mm) = 0._r8
+
+               do lspec = 1, nspec_amode(imode)
+                  mm = mam_idx(imode,lspec)
+                  raercol_nsav(kk,mm)    = raercol_nsav(kk,mm) + raercol_cw_nsav(kk,mm) ! cloud-borne aerosol
+                  raercol_cw_nsav(kk,mm) = 0._r8
+               enddo
+            enddo
+
+         endif  ! (cldn(icol,kk) > 0.01_r8) if-else structure
+
+      enddo  ! old_cloud_main_k_loop
```

Comment:
We can remove "! old_cloud_main_k_loop" comment.

Reply:
This is done.